comp/core/status/statusimpl/status.go

```diff
@@ -65,6 +65,10 @@ func newStatus(deps dependencies) (status.Component, error) {
 	collectorSectionPresent := false
 
 	for _, provider := range deps.Providers {
+		if _, ok := provider.(status.NoopProvider); ok {
```

Comment:
Can you explain why you need this dynamic check? It is often a sign that the interface (in this case `status.Provider`) has an issue.